### PR TITLE
Search related refactors

### PR DESCRIPTION
**Pull request by neelance**
No semantic changes.

### CODE REVIEW

**Comment by beyang** [comment]
LGTM

File: services/backend/internal/localstore/search/search_tok.go [hunk 1]
```diff
@@ -74,83 +74,49 @@ func ToTSVector(def *graph.Def) *TSVector {
 	if len(repoParts) >= 1 && (strings.HasSuffix(repoParts[0], ".com") || strings.HasSuffix(repoParts[0], ".org")) {
 		repoParts = repoParts[1:]
 	}
+	for _, w := range repoParts {
+		tsvector.Add("B", w, 1)
+	}
+	tsvector.Add("B", repoParts[len(repoParts)-1], 2) // the last path component tends to be the repository name
```
Comment (beyang) [inline services/backend/internal/localstore/search/search_tok.go]: needs a bounds check

File: services/backend/internal/localstore/search/search_tok.go [hunk 2]
```diff
@@ -74,83 +74,49 @@ func ToTSVector(def *graph.Def) *TSVector {
...
 		tsvector.Add("C", w, 1)
 	}
-	for i, w := range fileParts {
-		if len(fileParts)-1 == i {
-			tsvector.Add("C", w, 3)
-		} else {
-			tsvector.Add("C", w, 1)
-		}
-	}
+	tsvector.Add("C", fileParts[len(fileParts)-1], 2)
```
Comment (beyang) [inline services/backend/internal/localstore/search/search_tok.go]: bounds check

File: services/backend/internal/localstore/search/search_tok.go [hunk 3]
```diff
@@ -74,83 +74,49 @@ func ToTSVector(def *graph.Def) *TSVector {
...
+	for _, w := range repoParts {
+		tsvector.Add("B", w, 1)
+	}
+	tsvector.Add("B", repoParts[len(repoParts)-1], 2) // the last path component tends to be the repository name
+
 	unitParts := strings.Split(def.Unit, "/")
+	for _, w := range unitParts {
+		tsvector.Add("B", w, 1)
+	}
+	tsvector.Add("B", unitParts[len(unitParts)-1], 2)
```
Comment (beyang) [inline services/backend/internal/localstore/search/search_tok.go]: bounds check

File: services/backend/internal/localstore/search/search_tok.go (hunk 1 above)
Comment (neelance) [inline services/backend/internal/localstore/search/search_tok.go]: I don't think so. `strings.Split` never returns an empty slice.

File: services/backend/internal/localstore/search/search_tok.go (hunk 3 above)
Comment (neelance) [inline services/backend/internal/localstore/search/search_tok.go]: see above

File: services/backend/internal/localstore/search/search_tok.go (hunk 2 above)
Comment (neelance) [inline services/backend/internal/localstore/search/search_tok.go]: see above

**Pull request closed without merging**